swagger1/src/main/java/com/cerner/beadledom/swagger/SwaggerUiResource.java

```diff
@@ -44,11 +45,12 @@
   @GET
   @Path("/ui")
   @Produces(MediaType.TEXT_HTML)
-  public StreamingOutput getSwaggerUi(@Context UriInfo uriInfo) {
+  public StreamingOutput getSwaggerUi(@Context UriInfo uriInfo, @Context SecurityContext securityContext) {
+    String httpSchema = securityContext.isSecure() ? "https" : "http";
```

Comment:
Will this work behind a reverse proxy that terminates HTTPS? I think this would depend on if the SecurityContext understands the `X-Forwarded-Proto` header or other equivalent headers.

Reply:
`SecurityContext` doesn't understand those types of headers, but we can use `HttpHeaders` along with the `getRequestHeader()` method to determine if we are behind a reverse proxy. It might look a little something like this (without the tedious repetition of the common forms of the `X-Forwareded-Proto` header:
```java
public StreamingOutput getSwaggerUi(@Context UriInfo uriInfo,
      @Context SecurityContext securityContext, @Context HttpHeaders httpHeaders) {
    String httpScheme = isSecure() ? "https" : "http";
    return StreamingWriterOutput.with(writer -> MUSTACHE_FACTORY.compile("ui.mustache").execute(
        writer, ImmutableMap.of(
            "apiDocsUrl",
            uriInfo.getBaseUriBuilder().scheme(httpScheme).path(SwaggerApiResource.class).build().toASCIIString()))
    );
  }

  private Boolean isSecure(SecurityContext securityContext, HttpHeaders httpHeaders) {
    return securityContext.isSecure() ||
        !httpHeaders.getRequestHeader("X-Forwareded-Proto").isEmpty() ||
        !httpHeaders.getRequestHeader("Forwarded").isEmpty();
  }
```

Reply:
This is what i came up with, https://github.com/cerner/beadledom/pull/137/commits/8257840cfaa020ebc07620a2dc70c7cfabecb91e

I'm struggling to come up with a good way to test it, I tried putting it in the beadledom-client and running it but I really haven't come up with a good way of mocking `@Context`.